### ISSUE TITLE
Under some circumstances, the resolver fails to instantiate with JDK 11

### ISSUE THREAD

**Issue by ndw** [xmlresolver/xmlresolver#55]
See docbook/xslTNG#123 for example.

**Comment by CanOfBees** [comment]
@ndw I don't know if this is useful or not -- if there's something different I can try that would be helpful, please let me know. I installed several OpenJDKs and tried the [sampleapp ](https://github.com/xmlresolver/sampleapp/releases/tag/3.0.1) with the following:

```
bridger@ditrestle|/tmp/xmlresolver-sampleapp-3.0.1
) for JDK in $(find /usr/local/ -depth 1 -type d -name 'openjdk*'); do echo ${JDK}; ${JDK}/bin/java -jar sampleapp-3.0.1.jar parse xml/sample/invalid.xml; done
/usr/local/openjdk13
Performing a non-validating parse of xml/sample/invalid.xml
Exception in thread "main" java.lang.ClassCastException: class jdk.internal.loader.ClassLoaders$AppClassLoader cannot be cast to class java.net.URLClassLoader (jdk.internal.loader.ClassLoaders$AppClassLoader and java.net.URLClassLoader are in module java.base of loader 'bootstrap')
        at org.xmlresolver.XMLResolverConfiguration.findClasspathCatalogFiles(XMLResolverConfiguration.java:743)
        at org.xmlresolver.XMLResolverConfiguration.showConfig(XMLResolverConfiguration.java:613)
        at org.xmlresolver.XMLResolverConfiguration.loadConfiguration(XMLResolverConfiguration.java:315)
        at org.xmlresolver.XMLResolverConfiguration.<init>(XMLResolverConfiguration.java:211)
        at org.xmlresolver.example.SampleApp.getResolver(SampleApp.java:455)
        at org.xmlresolver.example.SampleApp.parse(SampleApp.java:167)
        at org.xmlresolver.example.SampleApp.process(SampleApp.java:108)
        at org.xmlresolver.example.SampleApp.main(SampleApp.java:74)
/usr/local/openjdk8
Performing a non-validating parse of xml/sample/invalid.xml
Using the XML Resolver with the following catalogs:
        jar:file:/tmp/xmlresolver-sampleapp-3.0.1/lib/xmlresolver-3.0.1-data.jar!/org/xmlresolver/catalog.xml
OASIS XML Catalogs processing instruction catalogs will be used
The resolver will not cache resources

✗ Resolved: book (file:///tmp/xmlresolver-sampleapp-3.0.1/xml/sample/invalid.xml)
Parse complete
/usr/local/openjdk15
Performing a non-validating parse of xml/sample/invalid.xml
Exception in thread "main" java.lang.ClassCastException: class jdk.internal.loader.ClassLoaders$AppClassLoader cannot be cast to class java.net.URLClassLoader (jdk.internal.loader.ClassLoaders$AppClassLoader and java.net.URLClassLoader are in module java.base of loader 'bootstrap')
        at org.xmlresolver.XMLResolverConfiguration.findClasspathCatalogFiles(XMLResolverConfiguration.java:743)
        at org.xmlresolver.XMLResolverConfiguration.showConfig(XMLResolverConfiguration.java:613)
        at org.xmlresolver.XMLResolverConfiguration.loadConfiguration(XMLResolverConfiguration.java:315)
        at org.xmlresolver.XMLResolverConfiguration.<init>(XMLResolverConfiguration.java:211)
        at org.xmlresolver.example.SampleApp.getResolver(SampleApp.java:455)
        at org.xmlresolver.example.SampleApp.parse(SampleApp.java:167)
        at org.xmlresolver.example.SampleApp.process(SampleApp.java:108)
        at org.xmlresolver.example.SampleApp.main(SampleApp.java:74)
/usr/local/openjdk16
Performing a non-validating parse of xml/sample/invalid.xml
Exception in thread "main" java.lang.ClassCastException: class jdk.internal.loader.ClassLoaders$AppClassLoader cannot be cast to class java.net.URLClassLoader (jdk.internal.loader.ClassLoaders$AppClassLoader and java.net.URLClassLoader are in module java.base of loader 'bootstrap')
        at org.xmlresolver.XMLResolverConfiguration.findClasspathCatalogFiles(XMLResolverConfiguration.java:743)
        at org.xmlresolver.XMLResolverConfiguration.showConfig(XMLResolverConfiguration.java:613)
        at org.xmlresolver.XMLResolverConfiguration.loadConfiguration(XMLResolverConfiguration.java:315)
        at org.xmlresolver.XMLResolverConfiguration.<init>(XMLResolverConfiguration.java:211)
        at org.xmlresolver.example.SampleApp.getResolver(SampleApp.java:455)
        at org.xmlresolver.example.SampleApp.parse(SampleApp.java:167)
        at org.xmlresolver.example.SampleApp.process(SampleApp.java:108)
        at org.xmlresolver.example.SampleApp.main(SampleApp.java:74)
/usr/local/openjdk11
Performing a non-validating parse of xml/sample/invalid.xml
Exception in thread "main" java.lang.ClassCastException: class jdk.internal.loader.ClassLoaders$AppClassLoader cannot be cast to class java.net.URLClassLoader (jdk.internal.loader.ClassLoaders$AppClassLoader and java.net.URLClassLoader are in module java.base of loader 'bootstrap')
        at org.xmlresolver.XMLResolverConfiguration.findClasspathCatalogFiles(XMLResolverConfiguration.java:743)
        at org.xmlresolver.XMLResolverConfiguration.showConfig(XMLResolverConfiguration.java:613)
        at org.xmlresolver.XMLResolverConfiguration.loadConfiguration(XMLResolverConfiguration.java:315)
        at org.xmlresolver.XMLResolverConfiguration.<init>(XMLResolverConfiguration.java:211)
        at org.xmlresolver.example.SampleApp.getResolver(SampleApp.java:455)
        at org.xmlresolver.example.SampleApp.parse(SampleApp.java:167)
        at org.xmlresolver.example.SampleApp.process(SampleApp.java:108)
        at org.xmlresolver.example.SampleApp.main(SampleApp.java:74)
```

All but OpenJDK8 fail, but some of this might be an OS quirk. I can try to get similar info from a Linux system, if that would be helpful. 

Thanks!

**Comment by ndw** [comment]
That is very useful. I'll see if I can sort it out tomorrow. Thanks!

**Comment by ndw** [comment]
I think this is fixed in 3.1.0 which is not yet referenced from the DocBook stylesheets.
